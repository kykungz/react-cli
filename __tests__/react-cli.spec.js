/* eslint-disable no-undef */

const reactCli = require('../src/react-cli')

describe.skip('create', () => {
  describe('class component', () => {
    test('should return correct template', () => {
      const flags = {
        class: true
      }
      const result = reactCli.create('MyComponent', flags)
      expect(result).toMatchSnapshot()
    })
    describe('with prop-types', () => {
      test('should return correct template', () => {
        const flags = {
          class: true,
          'prop-types': true
        }
        const result = reactCli.create('MyComponent', flags)
        expect(result).toMatchSnapshot()
      })
    })
  })

  describe('functional component', () => {
    test('should return correct template', () => {
      const flags = {
        functional: true
      }
      const result = reactCli.create('MyFunctionalComponent', flags)
      expect(result).toMatchSnapshot()
    })
  })

  describe('pure component', () => {
    test('should return correct template', () => {
      const flags = {
        pure: true
      }
      const result = reactCli.create('MyPureComponent', flags)
      expect(result).toMatchSnapshot()
    })
  })
})
