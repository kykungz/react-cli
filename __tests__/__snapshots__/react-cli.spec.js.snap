// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create class component should return correct template 1`] = `
"import React, { Component } from 'react'

class MyComponent extends Component {
  render () {
    return (
      <div>
      </div>
    )
  }
}

export default MyComponent"
`;

exports[`create class component with prop-types should return correct template 1`] = `
"
import React, { Component } from 'react'
import PropTypes from 'prop-types'

class MyComponent extends Component {
  render () {
    return (
      <div>
      </div>
    )
  }
}

MyComponent.propTypes = {}

export default MyComponent"
`;

exports[`create functional component should return correct template 1`] = `
"import React, { Component } from 'react'

class MyFunctionalComponent extends Component {
  render () {
    return (
      <div>
      </div>
    )
  }
}

export default MyFunctionalComponent"
`;

exports[`create pure component should return correct template 1`] = `
"import React, { Component } from 'react'

class MyPureComponent extends Component {
  render () {
    return (
      <div>
      </div>
    )
  }
}

export default MyPureComponent"
`;
